File: app/src/main/java/com/example/schedule/ui/LoginDialogFragment.java
package com.example.schedule.ui;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;

import com.example.schedule.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

public class LoginDialogFragment extends DialogFragment {

    private LoginActivity.RequestOperation operation;

    LoginDialogFragment(@NonNull LoginActivity.RequestOperation operation) {

        this.operation = operation;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();

        builder.setView(inflater.inflate(R.layout.fragment_login, null));

        return builder.create();
    }

    @Override
    public void onCancel(@NonNull DialogInterface dialog) {
        if (operation != null) operation.cancelRequest();

    }
}

File: app/src/main/java/com/example/schedule/ui/LoginDialogFragment.kt
package com.example.schedule.ui

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.example.schedule.R

class LoginDialogFragment internal constructor() : DialogFragment() {

    var actionAfterCancellation: () -> Unit = {}

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        val inflater = requireActivity().layoutInflater
        builder.setView(inflater.inflate(R.layout.fragment_login, null))
        return builder.create()
    }

    override fun onCancel(dialog: DialogInterface) {
        actionAfterCancellation()
        super.onCancel(dialog)
    }
}
